simulation_engine.py

```diff
@@ -0,0 +1,119 @@
+import lienard_wiechert
+from lienard_wiechert import SPEED_OF_LIGHT as SPEED_OF_LIGHT
+import kinematics.motion_lib as motion_lib
+
+import argparse
+import math
+import numpy as np
+import time
+import threading
+
+# radiation that has travelled further than this won't be plotted to save CPU and RAM
+MAX_DISTANCE = 15.0
+
+class VectorFieldSample(object):
+	def __init__(self, tail_position, vector = np.array([0.0, 0.0, 0.0])):
+		self.tail_position = tail_position
+		self.vector = vector
+
+
+class ElectricFieldSample(VectorFieldSample):
+	electric_field_function = None
```

Comment:
Because I want every ElectricFieldSample to use the same the same electric_field_function. When we advance the ElectricFieldSample instance by duration dt, the sample moves away from its original source a distance SPEED_OF_LIGHT * dt and the electric field vector gets updated based on the chosen formula.